### DISCUSSION TITLE
[PHP] Warning: Undefined variable~ 處理問題

### DISCUSSION THREAD

**Discussion by EshauHuang** [Lidemy/forum#153] [後端相關問題] [answered]
源於之前在使用 XAMPP 時，都不會跳錯誤訊息，一直以為是因為我用的是 install 版而不是 vim 版，但今天查了以後才發現是 php.ini 內的 display_errors 參數被關閉了，打開以後才發現之前覺得沒問題的網頁跳出很多錯誤。

主要跳的錯誤是這一條 `Warning: Undefined variable....`，而其實原因很簡單，就是因爲變數沒有宣告，例如我有一個變數 `$is_load`，需要登入狀態才會給一個值 `true`，而會影響到一些功能是否顯示，但因為是某些狀況才會給值，沒有提前宣告，因此才會跳出該錯誤。

在網路上有查到一些可以規避錯誤訊息的方式，而我繳交作業至 FileZilla 顯示的網頁都不會跳這些錯誤訊息，是不是代表我還是可以用例如 `if($is_load) {...}` 的語法來寫，然後在前加個＠變成 `if(@$is_load) {...}` 或是去修改 php.ini 來規避錯誤。我想問的是這樣的處理方法是可行的嗎？會不會之後有其它的問題，或者是還是建議使用 `empty` 或 `isset` 或在開頭先宣告此變數為NUll 會比較好？

謝謝各位！

**Comment by aszx87410** [comment, marked as answer]
這其實嚴格來講不是錯誤，它是 Warning，意思就是警告，提醒你說這樣寫可能不太好
不太好的原因是有可能會造成 bug，所以給你警告讓你提早避免掉，因此建議的做法是要用之前用 empty 或 isset 來檢查或是給個預設值，把警告消除掉

**Reply by EshauHuang** [reply]
感謝老師，之後還是嚴謹一點把所有變數都用 empty 或 isset 來檢查，來避免可能會出現的問題。
